server/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -164,6 +191,14 @@ public XContentBuilder(XContent xContent, OutputStream bos, Set<String> includes
     public XContentBuilder(XContent xContent, OutputStream os, Set<String> includes, Set<String> excludes) throws IOException {
         this.bos = os;
         this.generator = xContent.createGenerator(bos, includes, excludes);
+        Map<Class<?>, Writer> writers = new HashMap<>();
+        Map<Class<?>, HumanReadableTransformer> transformers = new HashMap<>();
+        for (XContentBuilderProvider service : ServiceLoader.load(XContentBuilderProvider.class)) {
```

Comment:
Can we do the service loading in a static block? Or during startup? It feels like we could still have one map if this were static. It wouldn't work with plugins but I think it'd work fine otherwise.

Reply:
This should definitely be static. Calling ServiceLoader.load is not "cheap" (must iterate classloaders/resources). The writers and transformers should be static members, and initialized in a static block which calls ServiceLoader. You can explicitly add writers and transformers for the known types there, in addition to the ones coming from the extensions.

Reply:
Okay, I'll move this to a static block for now, and move to passing it in at in a different place when the separation is made.